### PR TITLE
feat: added verify uniqueID of certificate

### PR DESCRIPTION
**Pull request by gyuguen**
### Implement
* Added `unique_id` field in `UnsignedCertificate`
* Added verify uniqueID of certificate
* Added test msg validation of deal

### CODE REVIEW

File: x/datadeal/types/message_deal.go [hunk 1]
```diff
@@ -82,46 +82,6 @@ func (m *MsgSubmitConsent) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{oracleAddress}
 }
 
-func (m *Certificate) ValidateBasic() error {
```
Comment (gyuguen) [inline x/datadeal/types/message_deal.go:85]: I moved it to `x/datadeal/types/certificate.go`

File: x/datadeal/types/message_deal_test.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
+package types
```
Comment (inchori) [inline x/datadeal/types/message_deal_test.go:1]: Does has the reason to write this test code?

**Review by inchori — APPROVED** [review]
lgtm

File: x/datadeal/types/message_deal_test.go (hunk 2 above)
Comment (gyuguen) [reply x/datadeal/types/message_deal_test.go:1]: Because I wanted to do a single test for `ValidateBasic()`

**Review by audtlr24 — APPROVED** [review]
good

**Review by H4NLee — APPROVED** [review]
lgtm

**Pull request merged**